### PR TITLE
Add locale options to tsserver

### PR DESCRIPTION
**Pull request by zhengbli**
Currently the server is always using English diagnostics messages. This PR adds the support to use messages in other locales.

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -4554,4 +4554,70 @@ namespace ts {
...
+        // Either ways do not fail, and fallback to the English diagnostic strings.
+        if (!trySetLanguageAndTerritory(language, territory, errors)) {
+            trySetLanguageAndTerritory(language, undefined, errors);
+        }
+
+        return true;
+    }
+
+    export function trySetLanguageAndTerritory(language: string, territory: string, errors?: Diagnostic[]): boolean {
+        const compilerFilePath = normalizePath(sys.getExecutingFilePath());
```
Comment (mhegazy) [inline src/compiler/utilities.ts]: make sys and argument here.

**Comment by mhegazy** [comment]
also this will need to be ported to release-2.1

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -4574,14 +4574,14 @@ namespace ts {
...
-            trySetLanguageAndTerritory(language, undefined, errors);
+        if (!trySetLanguageAndTerritory(language, territory, sys, errors)) {
+            trySetLanguageAndTerritory(language, /*territory*/ undefined, sys, errors);
         }
 
         return true;
     }
 
-    export function trySetLanguageAndTerritory(language: string, territory: string, errors?: Diagnostic[]): boolean {
+    export function trySetLanguageAndTerritory(language: string, territory: string, sys: System, errors?: Diagnostic[]): boolean {
```
Comment (mhegazy) [inline src/compiler/utilities.ts]: Do not think you need the whole thing..

consider : `host:  {getExecutingFilePath():string, readFile(name:string):string; }`

File: src/server/server.ts [hunk 3]
```diff
@@ -577,6 +577,11 @@ namespace ts.server {
         }
     }
 
+    const localeStr = findArgument("--locale");
+    if (localeStr && validateLocaleAndSetLanguage(localeStr, sys)) {
+        validateLocaleAndSetLanguage(localeStr, sys);
```
Comment (vladima) [inline src/server/server.ts]: not sure that I get why do we need to call `validateLocaleAndSetLanguage ` twice

**Comment by zhengbli** [comment]
@vladima Sorry that was a mistake, updated.

**Pull request merged**